src/sirhelpers.c

```diff
@@ -314,25 +314,28 @@ int _sir_fopen(FILE* restrict* restrict streamptr, const char* restrict filename
 
 struct tm* _sir_localtime(const time_t* restrict timer, struct tm* restrict buf) {
     if (_sir_validptr(timer) && _sir_validptr(buf)) {
-#if defined(__HAVE_STDC_SECURE_OR_EXT1__)
-# if defined(__WIN__)
-        errno_t ret = (errno_t)localtime_s(buf, timer);
+#if defined(__HAVE_STDC_SECURE_OR_EXT1__) && !defined(__EMBARCADEROC__)
```

Comment:
I think it would be better to put the embarcadero preproc where have stdc secure is defined in the first place, then we don't have to do these checks in the source

Reply:
It's done that way because their libc should have __HAVE_STDC_SECURE_OR_EXT1__ defined, except the localtime_s they provide doesn't work like the windows one, but they do provide a localtime_r, which Microsoft does not, and it's faster, like on Unix.